api/v3/Extension.php

```diff
@@ -346,9 +347,17 @@ function civicrm_api3_extension_get($params) {
     }
     $info = CRM_Extension_System::createExtendedInfo($obj);
     $info['id'] = $id++; // backward compatibility with indexing scheme
-    $result[] = $info;
+    if (!empty($params['key'])) {
```

Comment:
what about moving the filtering to the start of the loop - I have a feeling some of those calls are expensive